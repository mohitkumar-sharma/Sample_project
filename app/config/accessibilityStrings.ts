export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
