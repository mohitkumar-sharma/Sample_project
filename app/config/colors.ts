export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
