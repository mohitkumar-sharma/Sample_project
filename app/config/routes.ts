export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
