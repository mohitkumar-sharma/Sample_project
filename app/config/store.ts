import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
