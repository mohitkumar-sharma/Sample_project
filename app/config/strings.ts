export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
