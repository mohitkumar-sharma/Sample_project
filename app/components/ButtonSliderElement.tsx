import React from 'react';
import { View, StyleSheet, StyleProp, ViewStyle, TextStyle } from 'react-native';
import config from '../config';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

export interface ButtonSliderElementProps {
  /* as a string of icon name for react native vector Icon element */
  iconName?: string;
  /* as an object of style props for main content View element*/
  iconContainerStyle?: StyleProp<ViewStyle>;
  /* as an object of style props for react native vector Icon element*/
  iconStyle?: StyleProp<TextStyle>;
}

/**
 * This is the ButtonSliderElement functional component for slider button
 * @param props : props are defined in ButtonSliderElementProps which can be passed to this component
 */
const ButtonSliderElement = (props: ButtonSliderElementProps): JSX.Element => {
  const { iconContainerStyle, iconName, iconStyle } = props;

  return (
    <View style={[styles.container, iconContainerStyle]}>
      {iconName && iconName !== '' ? (
        <SimpleLineIcons
          name={iconName}
          style={[styles.icon, iconStyle]}
          size={20}
          color={config.colors.white}
          testID={config.accessibilityStrings.SLIDER_BUTTON_ICON}
        />
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 55,
    width: 55,
    backgroundColor: config.colors.themeBlue,
    borderRadius: 8,
  },
  icon: { fontSize: 25, color: config.colors.white },
});

export default ButtonSliderElement;
