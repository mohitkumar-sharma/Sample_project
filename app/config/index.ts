import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
