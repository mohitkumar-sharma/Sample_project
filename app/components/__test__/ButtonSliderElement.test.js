import React from 'react';
import { render } from '@testing-library/react-native';
import ButtonSliderElement from '../ButtonSliderElement';
import config from '../../config';

describe('ButtonSliderElement', () => {
  const props = { iconContainerStyle: { height: 60, width: 60 }, iconStyle: { fontSize: 25 } };

  const iconName = config.strings.DIAMOND_ICON;

  it('matches snapshot', () => {
    expect(render(<ButtonSliderElement {...props} />)).toMatchSnapshot();
    expect(render(<ButtonSliderElement {...props} iconName={iconName} />)).toMatchSnapshot();
  });

  it('renders the icon', () => {
    const { queryByTestId } = render(<ButtonSliderElement {...props} iconName={iconName} />);
    expect(queryByTestId(config.accessibilityStrings.SLIDER_BUTTON_ICON)).toBeTruthy();
  });

  it('does not render icon', () => {
    const { queryByTestId } = render(<ButtonSliderElement {...props} />);
    expect(queryByTestId(config.accessibilityStrings.SLIDER_BUTTON_ICON)).toBeFalsy();
  });
});
