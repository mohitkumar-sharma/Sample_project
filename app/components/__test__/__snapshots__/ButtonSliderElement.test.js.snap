// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSliderElement matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#6EB1F7",
        "borderRadius": 8,
        "height": 55,
        "justifyContent": "center",
        "width": 55,
      },
      Object {
        "height": 60,
        "width": 60,
      },
    ]
  }
/>
`;

exports[`ButtonSliderElement matches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#6EB1F7",
        "borderRadius": 8,
        "height": 55,
        "justifyContent": "center",
        "width": 55,
      },
      Object {
        "height": 60,
        "width": 60,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
        },
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 25,
          },
          Object {
            "fontSize": 25,
          },
        ],
        Object {
          "fontFamily": "simple-line-icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="slider-button-icon"
  >
    ÓÅÉ
  </Text>
</View>
`;
